cppcache/src/ClientProxyMembershipID.cpp

```diff
@@ -57,11 +57,11 @@ ClientProxyMembershipID::~ClientProxyMembershipID() noexcept {
 
 ClientProxyMembershipID::ClientProxyMembershipID(
     std::string dsName, std::string randString, const char* hostname,
-    uint32_t hostAddr, uint32_t hostPort, const char* durableClientId,
-    const std::chrono::seconds durableClntTimeOut)
-    : m_hostAddrAsUInt32(hostAddr) {
+	const char* hostAddr, int len, uint32_t hostPort, const char* durableClientId,
```

Comment:
Why are we bothering to pass around anything, but a string and a port? The IP Address should only matter at the lowest levels and in a certain sense not at all. We don't care that "142.156.37.2" or "www.whatever.com". I am not sure I understand this code, but it seems pretty clear we don't want to continue dealing with IPs.

Reply:
AFAIK, ClientProxyMembershipID is used *only* for serializing into a handshake message to initiate a server connection.  The IP address(es) are baked into the protocol here, so I don't think you can get rid of the logic.

Reply:
Yup, it's needed by the protocol / message here unfortunately.